pages/index.tsx

```diff
@@ -1,156 +1,206 @@
-  isWindow();
-  localStorage.setItem(key, values);
-}
-
-/**
- *
- * @returns a string that contains individual ID for key in React component.
- */
-function generateKeyID(): string {
-  isWindow();
-  return crypto.randomUUID();
-}
-
-function getFullLocalStorage() {
-  isWindow()
-  
-  let array: any = [];
-  for (let [key, values] of Object.entries(localStorage)) {
-    if (key.startsWith('gift_')) {
-      values = JSON.parse(values);
-      array = array.concat({
-        name: values['name'],
-        gift: values['gift'],
-        keyID: values['keyID'],
-      });
-    }
-  }
-  return array;
-}
-
-/**
- *
- * @returns an individual ID with randomUUID. The individual ID looks like: gift_number-array*
- *
- * number-array* = crypto.randomUUID()
- *
- * return example: gift_150cd819-1502-4717-9c96-f7ca7b42d8bd
- */
-function generateID(): string {
-  isWindow();
-  return `gift_${crypto.randomUUID()}`;
-}
-
-/**
- *
- * @returns if Window is undefined, throws an error
- */
-
-function isWindow() {
-  if (typeof window === 'undefined')
-    throw new Error(
-      'Window was defined as undefined. LocalStorage could not be read.',
-    );
-  return true;
-}
-
-type FullLocalStorage = {
-  name?: string;
-  gift?: string;
-  keyID?: string;
+export type FullLocalStorage = {
+  name: string;
+  gift: string;
+  id?: string;
+  localStorageKeyID?: string;
+  createdDate: number;
 };
 
 export default function Home() {
   const [giftData, setGiftData] = useState<FullLocalStorage[]>([]);
-  useEffect(() => {
-    isWindow(); // checks if Window is not undefined, else throws an error
+  const [giftNameError, setGiftNameError] = useState<boolean>(false);
+  const [receiverError, setReceiverError] = useState<boolean>(false);
+  const [newReceiver, setNewReceiver] = useState<string>('');
+  const [newGiftName, setNewGiftName] = useState<string>('');
 
+  useEffect(() => {
     console.log('effect');
-    const fullLocalStorage = getFullLocalStorage();
-    setGiftData((previousValue) => previousValue.concat(fullLocalStorage));
+    let parsedGiftData: [] = JSON.parse(getLocalStorage('giftData'));
+    if (parsedGiftData === null) {
+      parsedGiftData = [];
+      setLocalStorage('giftData', JSON.stringify(parsedGiftData));
+    }
+    const gifts = sortGiftsOldestFirst(parsedGiftData);
+    setGiftData(gifts);
   }, []);
 
+  function handleReceiverChange(event: ChangeEvent<HTMLInputElement>) {
+    setNewReceiver(event.target.value);
+  }
+
+  function handleGiftNameChange(event: ChangeEvent<HTMLInputElement>) {
+    setNewGiftName(event.target.value);
+  }
+
   function handleSubmit() {
-    const giftName: string = (
-      document.getElementById('giftName') as HTMLInputElement
-    ).value;
-    const receiverName: string = (
-      document.getElementById('giftReceiver') as HTMLInputElement
-    ).value;
+    setGiftNameError(false);
+    setReceiverError(false);
+    let errorFound: boolean = false;
```

Comment:
Näin voi tehdä, mutta tuo `:boolean` ei ole tarpeellinen koska TypeScript osaa päätellä sen tuosta `false`sta minkä sijoitat funktioon.